packages/abi-gen/test/generate_test/contracts/AbiGenDummy.sol

```diff
@@ -0,0 +1,79 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.5;
+
+
+contract AbiGenDummy
+{
+
+    uint256 constant internal SOME_CONSTANT = 1234;
+    string constant internal REVERT_REASON = "VALIDATOR_ERROR";
+
+    function simplePureFunction ()
+        public
+        pure
+        returns (uint256 result)
+    {
+        return 1;
+    }
+
+    function simplePureFunctionWithInput (uint256 x)
+        public
+        pure
+        returns (uint256 sum)
+    {
+        return 1 + x;
+    }
+
+    function pureFunctionWithConstant ()
+        public
+        pure
+        returns (uint256 someConstant)
+    {
+        return SOME_CONSTANT;
+    }
```

Comment:
Let's add a case where we call `ecrecover` since we know that failed in the WASM EVM implementation.

Reply:
Any other edge-cases you encountered while implementing your EVM PR?

Reply:
None that I recall right now :)